### PR TITLE
modules: lvgl: make the LVGL module optional

### PR DESCRIPTION
**Pull request by faxe1008**
Make the LVGL module optional.

### CODE REVIEW

File: samples/modules/lvgl/accelerometer_chart/README.rst [hunk 1]
```diff
@@ -33,6 +33,12 @@ Requirements
 Building and Running
 ********************
 
+First, ensure the optional LVGL module is enabled and available:
+
+   .. code-block:: console
```
Comment (kartben) [inline samples/modules/lvgl/accelerometer_chart/README.rst:38]:
```suggestion
   .. code-block:: shell
```

I know this is kinda weird (and tons of occurrences will need to be updated in current docs eventually...) -- see the note here: https://docs.zephyrproject.org/latest/contribute/documentation/guidelines.html#code-and-command-examples about shell  vs. console.

File: samples/modules/lvgl/demos/README.rst [hunk 2]
```diff
@@ -26,6 +26,12 @@ Requirements
 Building and Running
 ********************
 
+First, ensure the optional LVGL module is enabled and available:
+
+   .. code-block:: console
```
Comment (kartben) [inline samples/modules/lvgl/demos/README.rst:31]: ditto

File: samples/subsys/display/lvgl/README.rst [hunk 3]
```diff
@@ -45,6 +45,12 @@ or
 Building and Running
 ********************
 
+First, ensure the optional LVGL module is enabled and available:
+
+   .. code-block:: console
```
Comment (kartben) [inline samples/subsys/display/lvgl/README.rst:50]: ditto

**Review by kartben — CHANGES_REQUESTED** [review]
Didn't realize that LVGL repo is 170MB so I guess it makes sense.
Just a minor request for change.

File: samples/modules/lvgl/accelerometer_chart/README.rst [hunk 4]
```diff
@@ -33,6 +33,12 @@ Requirements
...
+      west config manifest.group-filter +optional
```
Comment (pdgendt) [inline samples/modules/lvgl/accelerometer_chart/README.rst:39]: Narrow down to a project filter (ref https://github.com/zephyrproject-rtos/west/issues/696)

```suggestion
      west config manifest.project-filter -- +lvgl
```

**Review by pdgendt — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by jfischer-no — CHANGES_REQUESTED** [review]
This is too tightly coupled with Zephyr and should not be optional.

**Comment by faxe1008** [comment]
> This is too tightly coupled with Zephyr and should not be optional.

From the original PR #61505 the intention of the optional modules is:
```
The idea is to look at the optional manifest as an area for modules that work with Zephyr, but not needed directly by zephyr.
```
At least from my point of view the LVGL module ticks all of those boxes.

**Review by MaureenHelm — APPROVED** [review]

**Comment by nashif** [comment]
> This is too tightly coupled with Zephyr and should not be optional.

Can you elaborate how is this tightly coupled?

**Comment by jfischer-no** [comment]
> @jfischer-no
> 
> > This is too tightly coupled with Zephyr and should not be optional.
> 
> Can you elaborate how is this tightly coupled?

We have 11 shields and 12 board containing LVGL configuration, there is also three samples and one test for LVGL in Zephyr tree.  
From the [Integration as optional modules](https://docs.zephyrproject.org/latest/contribute/external.html#integration-as-optional-modules),
"There shall not be any direct dependency added in the Zephyr code tree (Git repository) and all sample or test code shall be maintained as part of the module."

Either this dependencies are removed from the Zephyr tree, or it must remain non-optional.
(Btw, the move to optional breaks user application and IMO should be documented in the migration-guide.)

**Comment by fabiobaltieri** [comment]
I think @jfischer-no argument make sense, there's quite few boards referring to this from different vendors, it'd be nice to leave it in by default and have one less step for new users to check out a graphic stack.

**Comment by faxe1008** [comment]
> We have 11 shields and 12 board containing LVGL configuration, there is also three samples and one test for LVGL in Zephyr tree.
> From the [Integration as optional modules](https://docs.zephyrproject.org/latest/contribute/external.html#integration-as-optional-modules),

Ack. I did not consider this depedency. Thanks for raising your concern 👍

**Pull request closed without merging**